### DISCUSSION TITLE
Enums are not autocompleted & gives error

### DISCUSSION THREAD

**Discussion by jd-solanki** [fastapi/sqlmodel#1489] [Questions] [answered]
### First Check

- [X] I added a very descriptive title to this issue.
- [X] I used the GitHub search to find a similar issue and didn't find it.
- [X] I searched the SQLModel documentation, with the integrated search.
- [X] I already searched in Google "How to X in SQLModel" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to SQLModel but to [Pydantic](https://github.com/samuelcolvin/pydantic).
- [X] I already checked if it is not related to SQLModel but to [SQLAlchemy](https://github.com/sqlalchemy/sqlalchemy).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
class EnumAccessStatus(Enum):
    PENDING = "pending"
    APPROVED = "approved"
    REJECTED = "rejected"
    REVOKED = "revoked"

class Temp(SQLModel):
    py_type: EnumAccessStatus
    access_status = Field(
        sa_column=Column(
            sqlAlchemyEnum(EnumAccessStatus)
        )
    )
```

### Description

- Create an instance of Temp class by `Temp()` and hit space between brackets to get autocompletion
- Even If we add parameter by our self `Temp(py_type=EnumAccessStatus.REJECTED, access_status=EnumAccessStatus.REJECTED)` it throws "No parameter named `access_status`"

### Operating System

Linux

### Operating System Details

Ubuntu 21.10 64-bit

### SQLModel Version

0.0.6

### Python Version

3.10.2

### Additional Context

![image](https://user-images.githubusercontent.com/47495003/159639061-ef9cf1dd-b70d-4afd-916d-093a62881e4a.png)
![image](https://user-images.githubusercontent.com/47495003/159639104-32107104-9af2-4d4e-ae6f-2c27e44d30f8.png)

**Comment by masreplay** [comment]
As mentioned in python docs you have to inherit from both `Enum` and the value that will set in orinal like `str`
```
class MyEnum(str, Enum):
    Pending= "pendding"
````

**Comment by jd-solanki** [comment]
It is still not working.

![image](https://user-images.githubusercontent.com/47495003/159845971-059252cd-3e34-4cf9-9c3b-5b2fa3276d93.png)

**Comment by YuriiMotov** [comment, marked as answer]
> it throws "No parameter named access_status"

This is because you didn't specify the type of this parameter.

`sa_column` is only for SQLAlchemy. You still need to add type annotations for this field for Pydantic to be able to create the init method:

```py
    access_status: EnumAccessStatus = Field(
        sa_column=Column(
            sqlAlchemyEnum(EnumAccessStatus)
        )
    )
```
